test(api): Static check for ActionController[#60]

akshaylg0314 opened the pull request:
📝 PR Description
Static check for ActionController

🔗 Related Issue
Closes https://github.com/eclipse-pullpiri/pullpiri/issues/60

🧪 Test Method
cargo fmt -- --check

cargo clippy

cargo deny check

📸 Screenshots
Cargo clippy output:
![Screenshot from 2025-05-14 13-17-39](https://github.com/user-attachments/assets/73169794-27a6-479c-ac5c-197abd2f6b9e)

Cargo deny check output:
![Screenshot from 2025-05-14 13-39-33](https://github.com/user-attachments/assets/212d8389-5793-45db-b26b-5029c7387461)

✅ Checklist
[✅] Code conventions are followed
[✅] Tests are added/modified
[X] Documentation is updated (if necessary)

youngtaekiim commented on src/common/Cargo.toml:
```diff
@@ -6,12 +6,13 @@
 [dependencies]
 const_format = "0.2.33"
-config = "0.14.1"
+config = "0.14.1" 
```
Please remove white space from the end of line

akshaylg0314 replied:
Done :https://github.com/eclipse-pullpiri/pullpiri/pull/65/commits/9734eadf8d2013002b2864fe352e1d43a629b86b

youngtaekiim commented on src/player/actioncontroller/Cargo.toml:
```diff
@@ -7,8 +7,8 @@ license = "Apache-2.0"
-tokio = { version = "1.36.0", features = ["full"] }
-tonic = "0.12.3"
+tokio = { version = "1.45.0", features = ["full"] }
+tonic = "0.12.3"  
```
Remove white space.

Don't use different versions of dependency crates.
If possible, use the same version of dependency for all pullpiri modules.
I sometimes test the Rust version and the dependency versions and then upgrade them all at once.

akshaylg0314 replied:
Done :https://github.com/eclipse-pullpiri/pullpiri/pull/65/commits/9734eadf8d2013002b2864fe352e1d43a629b86b

youngtaekiim replied:
I upload a commit to revert tokio version

akshaylg0314 replied:
I intentionally used tokio version 1.45.0 because there is a vulnerability reported for version 1.36.0 or lower in the Rust advisory database.

youngtaekiim replied:
> I intentionally used tokio version 1.45.0 because there is a vulnerability reported for version 1.36.0 or lower in the Rust advisory database.

Thanks for your new information.
Please make a new issue.
If you cannot create issue, ask to leeeunkoo (FL).

akshaylg0314 replied:
ok sure i will create new issue and deliver the changes

youngtaekiim commented on src/player/actioncontroller/deny.toml:
```diff
@@ -0,0 +1,238 @@
+# This template contains all of the possible sections and their default values
```
I didn't know much about `cargo deny`.
I'll study this part and apply it to other modules.
I'll resolve this conversation without further confirmation.

akshaylg0314 replied:
ok

youngtaekiim commented on src/player/actioncontroller/src/grpc/sender.rs:
```diff
@@ -27,6 +27,7 @@ use tonic::Request;
 /// - The connection to PolicyManager is not established
 /// - The gRPC request fails
 /// - The policy check fails
+#[allow(dead_code)]
```
Please do not use attributes like `allow(dead_code)`, `allow(unused_varialbe)` and so on.
I intentionally omitted the areas that require further review in order to quickly highlight them through warnings.

akshaylg0314 replied:
Done :https://github.com/eclipse-pullpiri/pullpiri/pull/65/commits/9734eadf8d2013002b2864fe352e1d43a629b86b

youngtaekiim commented on src/player/actioncontroller/src/manager.rs:
```diff
@@ -19,7 +19,7 @@ pub struct ActionControllerManager {
     // Add other fields as needed
 }
-
+#[allow(dead_code)]
```
Please do not use attributes like allow(dead_code), allow(unused_varialbe) and so on.
I intentionally omitted the areas that require further review in order to quickly highlight them through warnings.

akshaylg0314 replied:
Done :https://github.com/eclipse-pullpiri/pullpiri/pull/65/commits/9734eadf8d2013002b2864fe352e1d43a629b86b

youngtaekiim commented on src/player/actioncontroller/src/runtime/bluechi/mod.rs:
```diff
@@ -60,6 +60,7 @@ impl Command {
 /// * `scenario_name` - Name of the scenario to operate on
 /// * `node` - Name of the node to target
 /// * `bluechi_cmd` - The Bluechi command to execute
+#[allow(clippy::let_underscore_future)]
```
Please do not use attributes for cargo clippy lint.
I intentionally omitted the areas that require further review in order to quickly highlight them through warnings.

akshaylg0314 replied:
Done :https://github.com/eclipse-pullpiri/pullpiri/pull/65/commits/9734eadf8d2013002b2864fe352e1d43a629b86b

youngtaekiim commented on src/player/actioncontroller/src/runtime/mod.rs:
```diff
@@ -17,6 +17,7 @@ pub mod nodeagent;
 /// Returns an error if:
 /// - Configuration for either runtime system is invalid
 /// - Connection to runtime systems fails
+#[allow(dead_code)]
```
Please do not use attributes like allow(dead_code), allow(unused_varialbe) and so on.
I intentionally omitted the areas that require further review in order to quickly highlight them through warnings.

akshaylg0314 replied:
Done :https://github.com/eclipse-pullpiri/pullpiri/pull/65/commits/9734eadf8d2013002b2864fe352e1d43a629b86b

youngtaekiim commented on src/player/actioncontroller/src/runtime/nodeagent/mod.rs:
```diff
@@ -1,18 +1,20 @@
+#![allow(unused_variables)]
```
Please do not use attributes like allow(dead_code), allow(unused_varialbe) and so on.
I intentionally omitted the areas that require further review in order to quickly highlight them through warnings.

akshaylg0314 replied:
Done :https://github.com/eclipse-pullpiri/pullpiri/pull/65/commits/9734eadf8d2013002b2864fe352e1d43a629b86b

youngtaekiim commented on src/player/actioncontroller/src/runtime/nodeagent/mod.rs:
```diff
@@ -1,18 +1,20 @@
+#![allow(unused_variables)]
 use common::Result;
 use std::collections::HashMap;
-
```
revert this line.
Please insert new line before struct or method.

akshaylg0314 replied:
Done :https://github.com/eclipse-pullpiri/pullpiri/pull/65/commits/9734eadf8d2013002b2864fe352e1d43a629b86b

youngtaekiim commented:
Dear @akshaylg0314 ,
I left some minor comments, so I'd like to request additional commits for them.
Thanks.

akshaylg0314 commented:
Hi @youngtaekiim ,Thanks for your review 
I want to ask should we skip cargo clippy check as of now for static check? Because without using allow dead or other attributes clippy lint will fail.

youngtaekiim commented:
> Hi @youngtaekiim ,Thanks for your review I want to ask should we skip cargo clippy check as of now for static check? Because without using allow dead or other attributes clippy lint will fail.

Since fast releases are more important, we don't need to focus on fixing warnings or errors in `cargo clippy`. (We just need to make sure the build succeeds.)
I don't know the exact release schedule, but if you're curious, please contact `edo.lee (leeeunkoo)` .
We'll look at warnings in cargo clippy and rust-analyzer one by one later.

Thanks for your interest.

akshaylg0314 commented:
Ok t

> > Hi @youngtaekiim ,Thanks for your review I want to ask should we skip cargo clippy check as of now for static check? Because without using allow dead or other attributes clippy lint will fail.
> 
> Since fast releases are more important, we don't need to focus on fixing warnings or errors in `cargo clippy`. (We just need to make sure the build succeeds.) I don't know the exact release schedule, but if you're curious, please contact `edo.lee (leeeunkoo)` . We'll look at warnings in cargo clippy and rust-analyzer one by one later.
> 
> Thanks for your interest.

Ok then we will ignore warnings as of now in cargo clippy and i will make one more commit by addressing all your review comments
Thanks!!

youngtaekiim reviewed (approved):
Very good job.
